File: backend/routers/resumes.py
import os
import shutil
from fastapi import APIRouter, UploadFile, File, HTTPException
from fastapi.responses import FileResponse

UPLOAD_DIR = "uploads"
os.makedirs(UPLOAD_DIR, exist_ok=True)

router = APIRouter(prefix="/resumes", tags=["resumes"])

@router.post("/upload")
async def upload_resume(file: UploadFile = File(...)):
    if file.content_type != "application/pdf":
        raise HTTPException(status_code=400, detail="Only PDF files allowed")
    
    file_path = os.path.join(UPLOAD_DIR, file.filename)
    with open(file_path, "wb") as buffer:
        shutil.copyfileobj(file.file, buffer)
    
    return { "filename": file.filename, "path": file_path }

@router.get("/")
def list_resumes():
    files = os.listdir(UPLOAD_DIR)
    return [
        { "filename": f, "size": os.path.getsize(os.path.join(UPLOAD_DIR, f)) }
        for f in files if f.endswith(".pdf")
    ]

@router.delete("/{filename}")
def delete_resume(filename: str):
    file_path = os.path.join(UPLOAD_DIR, filename)
    if not os.path.exists(file_path):
        raise HTTPException(status_code=404, detail="File not found")
    os.remove(file_path)
    return { "message": "Deleted" }

File: frontend/src/api/resumes.js
import axios from "axios"

const BASE = "http://localhost:8000/resumes"

export async function uploadResume(file) {
    const formData = new FormData()
    formData.append("file", file)
    const res = await axios.post(`${BASE}/upload`, formData)
    return res.data
}

export async function listResumes() {
    const res = await axios.get(`${BASE}/`)
    return res.data
}

export async function deleteResume(filename) {
    const res = await axios.delete(`${BASE}/${filename}`)
    return res.data
}